Device/ARM/ARMCR52/startup_ARMCR52.c

```diff
@@ -0,0 +1,159 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if defined (ARMCR52)
+  #include "ARMCR52.h"
+#else
+  #error device not specified!
+#endif
+
+/*----------------------------------------------------------------------------
+  Linker generated Symbols
+ *----------------------------------------------------------------------------*/
+extern uint32_t __HypStackTop;
+extern uint32_t __ecc_init_start__, __ecc_init_end__;
+
+/*----------------------------------------------------------------------------
+  External References
+ *----------------------------------------------------------------------------*/
+extern uint32_t __INITIAL_SP;
+extern uint32_t __STACK_LIMIT;
+
+extern void __PROGRAM_START(void) __NO_RETURN;
+
+/*----------------------------------------------------------------------------
+  Internal References
+ *----------------------------------------------------------------------------*/
+void EL2_Default_Handler(void) __NO_RETURN;
+void EL1_Default_Handler(void) __NO_RETURN;
+void EL2_Reset_Handler    (void) __attribute__((naked,__noreturn__));
+
+/*----------------------------------------------------------------------------
+  Exception / Interrupt Handler
+ *----------------------------------------------------------------------------*/
+void EL2_Undef_Handler    (void) __attribute__ ((weak, alias("EL2_Default_Handler")));
+void EL2_SVC_Handler      (void) __attribute__ ((weak, alias("EL2_Default_Handler")));
+void EL2_Prefetch_Handler (void) __attribute__ ((weak, alias("EL2_Default_Handler")));
+void EL2_Abort_Handler    (void) __attribute__ ((weak, alias("EL2_Default_Handler")));
+void EL2_Reserved_Handler (void) __attribute__ ((weak, alias("EL2_Default_Handler")));
+void EL2_IRQ_Handler      (void) __attribute__ ((weak, alias("EL2_Default_Handler")));
+void EL2_FIQ_Handler      (void) __attribute__ ((weak, alias("EL2_Default_Handler")));
+
+void EL1_Reset_Handler    (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+void EL1_Undef_Handler    (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+void EL1_SVC_Handler      (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+void EL1_Prefetch_Handler (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+void EL1_Abort_Handler    (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+void EL1_Reserved_Handler (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+void EL1_IRQ_Handler      (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+void EL1_FIQ_Handler      (void) __attribute__ ((weak, alias("EL1_Default_Handler")));
+
+/*----------------------------------------------------------------------------
+  Exception / Interrupt Vector table
+ *----------------------------------------------------------------------------*/
+void __VECTOR_TABLE_ATTRIBUTE __EL2_Vectors(void) {
+  __ASM volatile(
```

Comment:
This again is not really compiler agnostic as it uses GNU assembly syntax. Any chance to have this in C?

Reply:
I don't have other compilers right on my plate. Do you have an example ?